src/client/entity-editor/common/linked-entity.js

```diff
@@ -20,50 +20,63 @@
 // @flow
 
 // import Icon from 'react-fontawesome';
+import FontAwesome from 'react-fontawesome';
 import React from 'react';
 import {genEntityIconHTMLElement} from '../../helpers/entity';
 
 
 type LinkedEntityProps = {
-	disambiguation?: ?string,
-	id?: string | false,
-	text: string,
-	type: string,
-	unnamedText?: string
+	children: PropTypes.node,
+	className: PropTypes.string,
+	isDisabled: PropTypes.bool,
+	isFocused: PropTypes.bool,
+	innerProps: any,
+	isSelected: PropTypes.bool,
+	onFocus: PropTypes.func,
+	onSelect: PropTypes.func,
+	option: PropTypes.object.isRequired,
 };
 
 function LinkedEntity(
-	{disambiguation, id, text, type, unnamedText}: LinkedEntityProps
+	{onSelect, option}: LinkedEntityProps
 ) {
+	const {disambiguation, id, text, type, unnamedText, language} = option;
 	let url = null;
 	if (type) {
 		url = type === 'Area' ?
 			`//musicbrainz.org/area/${id}` :
 			`/${type.toLowerCase()}/${id}`;
 	}
-	function eventHandler(event) {
-		// event.preventDefault();
-		// console.log(url, 'url', text);
+	function parentEventHandler(event) {
+		onSelect(option, event);
+	}
+	function childEventHandler(event) {
+		event.stopPropagation();
+		event.preventDefault();
 		if (url) {
 			window.open(url, '_blank');
 		}
 	}
 
 	const nameComponent = text || <i>{unnamedText}</i>;
 	const contents = (
-		<span>
+		<div className="react-select__value-container" style={{cursor: 'pointer', fontSize: '15px', padding: '8px'}} onClick={parentEventHandler}>
```

Comment:
The className made me cringe. That's not a right way to go about it.
Thankfully, if you look at LinkedEntityProps, classname is passed down, so you can add it to the arguments along with `{onSelect, option}` line 41,  and then pass it in this div. Don't forget to remove the `style` as well.

Reply:
Okay